### ISSUE TITLE
How to train my dataset

### ISSUE THREAD

**Issue by wenshihao1** [davidnvq/grit#45]
I don't know how to start training my dataset. Can you tell me the steps?

**Comment by quangvnai** [comment]
May you provide the structure of the annotations you have?
BTW, I have an example to train on custom dataset here: https://github.com/davidnvq/grit#finetune--retrain-grit-on-your-own-dataset

**Comment by wenshihao1** [comment]
export DATA_ROOT=path/to/coco_dataset
# with pretrained object detector on 4 datasets
python train_caption.py exp.name=caption_4ds model.detector.checkpoint=4ds_detector_path

# with pretrained object detector on Visual Genome
python train_caption.py exp.name=caption_4ds model.detector.checkpoint=vg_detector_path


this i can't unsderstand.  "exp.name=caption_4ds model.detector.checkpoint=4ds_detector_path"  what's means?

**Comment by wenshihao1** [comment]
#vi_caption_path = "/media/local_workspace/quang/datasets/vietcap/vi_captions.json"
This means all data in json format?

**Comment by wenshihao1** [comment]
Sorry, i can't find the example to train my dataset. I just can find three files.
Thanks for your reply.
Good luck with your work


















At 2023-08-01 21:07:27, "Van-Quang Nguyen" ***@***.***> wrote:

May you provide the structure of the annotations you have?
BTW, I have an example to train on custom dataset here: https://github.com/davidnvq/grit#finetune--retrain-grit-on-your-own-dataset

**Comment by Wangdanchunbufuz** [comment]
How do I label my own data set?Is there any tool to get the caption?I look forward to your answer。

**Comment by quangvnai** [comment]
@Wangdanchunbufuz Hi there, may I get in detail what format of your dataset now?

**Comment by Wangdanchunbufuz** [comment]
> @Wangdanchunbufuz Hi there, may I get in detail what format of your dataset now?

I'm sorry for the late reply！If I want to annotate my own data set to train your model, should I annotate the mscoco dataset in json format? Is there any annotation software that can be annotated in json format? Looking forward to your reply！

**Comment by Dufresue** [comment]
> > @Wangdanchunbufuz Hi there, may I get in detail what format of your dataset now?
> 
> I'm sorry for the late reply！If I want to annotate my own data set to train your model, should I annotate the mscoco dataset in json format? Is there any annotation software that can be annotated in json format? Looking forward to your reply！

have you figured out how to prepare custom dataset?
